tests/integration/features/istio/production/main-suite/istio_configuration.feature

```diff
@@ -61,3 +61,10 @@ Feature: Configuration of Istio module
     And Log of container "ext-authz" in deployment "ext-authz" in namespace "default" contains "X-Add-In-Check:[value] X-Ext-Authz:[allow]"
     And Request with header "x-ext-authz" with value "deny" to path "/headers" should have response code "403"
     And Log of container "ext-authz" in deployment "ext-authz" in namespace "default" contains "X-Add-In-Check:[value] X-Ext-Authz:[deny]"
+  Scenario: Compatibility mode is on
```

Comment:
I don't like that we have tests that we need to update on every Istio version update, but there might be no good way to observe a different behaviour for compatibilityVersion. I'd also have this as a separate feature and not add it to the Istio configuration. Also some of those compatibilityVersion changes might not be observable by checking ENV Variables, what do we do then? An example for this is the [default tracing changes in 1.22](https://istio.io/latest/news/releases/1.22.x/announcing-1.22/upgrade-notes/#default-tracing-to-zipkinistio-systemsvc-removed). In this case this test would be completely invalid if only such changes are controlled by the compatibilityVersion.

Reply:
We might want to update this tests to verify that those env vars are not sent before we switch the compatibility mode one and then verify that they are set. This will require to update this test with every Istio release. I think there is no other way to have the verification that the flag we provide to Istio works as expected. 
Also there might be the case where the compatibility version doesn't use any env vars, so this test might not be applicable.

Reply:
as we discussed, let's go with this approach